magneticfield/inc/fieldPropagatorConstBz.h

```diff
@@ -11,110 +11,65 @@
 #include <AdePT/BlockData.h>
 #include <AdePT/LoopNavigator.h>
 
-#include "track.h"
 #include "ConstBzFieldStepper.h"
 
-#include "IterationStats.h"
-#ifdef CHORD_STATS
-extern IterationStats *chordIterStatsPBz;
-extern __device__ IterationStats_impl *chordIterStatsPBz_dev;
-#endif
-
 // Data structures for statistics of propagation chords
 
 class fieldPropagatorConstBz {
 public:
   __host__ __device__ fieldPropagatorConstBz(float Bz) { BzValue = Bz; }
   __host__ __device__ ~fieldPropagatorConstBz() {}
 
-  __host__ __device__ void stepInField(track &aTrack, vecgeom::Vector3D<double> &endPosition,
-                                       vecgeom::Vector3D<double> &endDirection);
+  __host__ __device__ void stepInField(double kinE, double mass, int charge, double step,
+                                       vecgeom::Vector3D<double> &position, vecgeom::Vector3D<double> &direction);
 
   template <bool Relocate = true>
-  __device__ //  __host__
-      double
-      ComputeStepAndPropagatedState(track &aTrack, float physicsStep);
+  __host__ __device__ double ComputeStepAndPropagatedState(double kinE, double mass, int charge, double physicsStep,
+                                                           vecgeom::Vector3D<double> &position,
+                                                           vecgeom::Vector3D<double> &direction,
+                                                           vecgeom::NavStateIndex const &current_state,
+                                                           vecgeom::NavStateIndex &new_state);
 
 private:
   float BzValue;
 };
 
-// Cannot make __global__ method part of class
-__global__ void moveInField(adept::BlockData<track> *trackBlock, fieldPropagatorConstBz &fieldProp);
-//  float                     BzValue );
-
 constexpr double kPushField = 1.e-8;
 
 // -----------------------------------------------------------------------------
 
-__host__ __device__ void fieldPropagatorConstBz::stepInField(track &aTrack,
-                                                             // float      BzValue,
-                                                             vecgeom::Vector3D<double> &endPosition,
-                                                             vecgeom::Vector3D<double> &endDirection)
+__host__ __device__ void fieldPropagatorConstBz::stepInField(double kinE, double mass, int charge, double step,
+                                                             vecgeom::Vector3D<double> &position,
+                                                             vecgeom::Vector3D<double> &direction)
 {
-  using copcore::units::kElectronMassC2;
-
-  double step = aTrack.interaction_length;
-  int charge  = aTrack.charge();
-
-  if (charge != 0.0) {
-    double kinE        = aTrack.energy;
-    double momentumMag = sqrt(kinE * (kinE + 2.0 * kElectronMassC2));
-    // aTrack.mass() -- when extending with other charged particles
+  if (charge != 0) {
```

Comment:
I understand that using mass simplifies the code, but think that it can be kept as the constant 'kElectronMassC2' by adding an assert.